packages/openjdk24.rb

```diff
@@ -0,0 +1,49 @@
+require 'package'
+
+class Openjdk24 < Package
+  description 'The JDK is a development environment for building applications, applets, and components using the Java programming language.'
+  homepage 'https://openjdk.org/'
+  version '24.0.0'
+  license 'GPL-2'
+  compatibility 'x86_64'
+  # Visit https://www.azul.com/downloads/?package=jdk#zulu to download the binary.
+  source_url 'https://cdn.azul.com/zulu/bin/zulu24.28.83-ca-jdk24.0.0-linux_x64.tar.gz'
+  source_sha256 '29fea017c03c645221ba31208e510d7ae48f5735ba4169d565c46cb77e7f66f2'
+
+  no_compile_needed
+  no_shrink
+  print_source_bashrc
+
+  def self.preflight
+    jdkver = `java -version 2>&1`[/version "(.*?)"/, 1].to_s
+    unless jdkver.empty? || jdkver.include?('No such file or directory') || jdkver.include?('not found')
+      jdkname = `java -version 2>&1`[/(.*?)\s/, 1].to_s
+      jdkname = 'jdk' if jdkname == 'java'
+      majver = jdkver[/(\d+)\./, 1].to_s
+      majver = '8' if majver == '1'
+      unless jdkname == 'openjdk' && majver == '17'
+        puts "Package #{jdkname}#{majver} already installed.".lightgreen
+        abort "Enter `crew remove #{jdkname}#{majver} && crew install openjdk21` to install this version."
+      end
+    end
+  end
+
+  def self.install
+    FileUtils.mkdir_p CREW_DEST_MAN_PREFIX
+    FileUtils.mkdir_p "#{CREW_DEST_PREFIX}/share/openjdk23"
```

Comment:
```suggestion
    FileUtils.mkdir_p "#{CREW_DEST_PREFIX}/share/openjdk24"
```

and then the same change for all the other mentions of openjdk23 here